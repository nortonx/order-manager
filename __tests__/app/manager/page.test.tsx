import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "@jest/globals";
import OrderManagerPage from "@/app/manager/page";

describe("AssetsManagerPage", () => {
  it("renders the AssetsManager page", async () => {
    const { container } = render(<OrderManagerPage />);
    const orderManagerElement = screen.getByTestId("manager-page");
    expect(orderManagerElement).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
