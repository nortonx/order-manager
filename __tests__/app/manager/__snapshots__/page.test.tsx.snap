// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetsManagerPage renders the AssetsManager page 1`] = `
<div>
  <div
    class="w-full border rounded-sm p-2"
  >
    <h1
      class="font-bold text-2xl text-italic border-b-slate-300 p-4"
    >
      Gerenciador de Ordens
    </h1>
    <div
      class="grid grid-cols-12 gap-2 w-full"
      data-testid="manager-page"
    >
      <div
        class="col-span-10 w-full bg-white dark:bg-slate-800 rounded-lg shadow"
        data-testid="orders-table"
      >
        <p
          class="max-w-fit mx-auto p-4 font-bold text-xl"
        >
          Nenhum resultado listado
        </p>
      </div>
      <div
        class="col-span-2 border border-slate-800 bg-white dark:bg-slate-800 rounded-lg shadow"
        data-testid="manager-panel"
      >
        <form
          class="p-4"
        >
          <fieldset>
            <legend
              class="font-bold py-2 underline"
            >
              Busca por ativos
            </legend>
            <div
              data-testid="asset-form-fields"
            >
              <div
                class="mb-4"
              >
                <label
                  class="block text-sm font-medium text-gray-700 dark:text-gray-300 mb-2"
                >
                  Instrumento
                </label>
                <input
                  aria-invalid="false"
                  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive my-2"
                  data-slot="input"
                  name="symbol"
                  placeholder="Instrumento"
                  value=""
                />
              </div>
              <div
                class="mb-4"
              >
                <label
                  class="block text-sm font-medium text-gray-700 dark:text-gray-300 mb-2"
                >
                  Tipo de Operação
                </label>
                <div
                  class="select-container"
                >
                  <button
                    aria-autocomplete="none"
                    aria-controls="radix-«r0»"
                    aria-expanded="false"
                    aria-invalid="false"
                    class="border-input data-[placeholder]:text-muted-foreground [&_svg:not([class*='text-'])]:text-muted-foreground focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive dark:bg-input/30 dark:hover:bg-input/50 flex items-center justify-between gap-2 rounded-md border bg-transparent px-3 py-2 text-sm whitespace-nowrap shadow-xs transition-[color,box-shadow] outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50 data-[size=default]:h-9 data-[size=sm]:h-8 *:data-[slot=select-value]:line-clamp-1 *:data-[slot=select-value]:flex *:data-[slot=select-value]:items-center *:data-[slot=select-value]:gap-2 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 w-full"
                    data-size="default"
                    data-slot="select-trigger"
                    data-state="closed"
                    dir="ltr"
                    role="combobox"
                    type="button"
                  >
                    <span
                      data-slot="select-value"
                      style="pointer-events: none;"
                    >
                      Compra
                    </span>
                    <svg
                      aria-hidden="true"
                      class="lucide lucide-chevron-down size-4 opacity-50"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m6 9 6 6 6-6"
                      />
                    </svg>
                  </button>
                  <select
                    aria-hidden="true"
                    name="type"
                    style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
                    tabindex="-1"
                  >
                    <option
                      selected=""
                      value="compra"
                    >
                      Compra
                    </option>
                    <option
                      value="venda"
                    >
                      Venda
                    </option>
                  </select>
                </div>
              </div>
              <div
                class="mb-4"
              >
                <label
                  class="block text-sm font-medium text-gray-700 dark:text-gray-300 mb-2"
                >
                  Quantidade
                </label>
                <input
                  aria-invalid="false"
                  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive my-2"
                  data-slot="input"
                  max="10000"
                  min="1"
                  name="quantity"
                  placeholder="Quantidade"
                  type="number"
                  value="1"
                />
              </div>
              <div
                class="mb-4"
              >
                <label
                  class="block text-sm font-medium text-gray-700 dark:text-gray-300 mb-2"
                >
                  Preço Total
                </label>
                <input
                  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground border-input flex h-9 w-full min-w-0 rounded-md border px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive my-1 bg-gray-50 dark:bg-gray-800"
                  data-slot="input"
                  disabled=""
                  name="totalPrice"
                  placeholder="Preço"
                  readonly=""
                  type="text"
                  value="R$ 0,00"
                />
              </div>
            </div>
          </fieldset>
          <div
            class="action-buttons flex justify-between"
          >
            <button
              class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-8 rounded-md gap-1.5 px-3 has-[>svg]:px-2.5"
              data-slot="button"
              disabled=""
              type="submit"
            >
              Adicionar
            </button>
            <button
              class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-8 rounded-md gap-1.5 px-3 has-[>svg]:px-2.5"
              data-slot="button"
              type="button"
            >
              Cancelar
            </button>
          </div>
          <ul
            class="filtered-assets-list mt-4 text-sm"
            data-testid="asset-search-results"
          >
            <li>
              Nenhum resultado listado
            </li>
          </ul>
        </form>
      </div>
    </div>
  </div>
</div>
`;
